### ISSUE TITLE
u8 Migration: ConvertTinyMceAndGridMediaUrlsToLocalLink fails on Umbraco 8.6.0

### ISSUE THREAD

**Issue by robertjf** [umbraco/Umbraco-CMS#7816]
Although measures have been taken to ensure the queries for generating the sql to read the property details using the `PropertyTypeDto80` and friends works to exclude the version 8.6 columns `mandatoryMessage` and `validationRegExpMessage`, line 105 in ConvertTinyMceAndGridMediaUrlsToLocalLink.cs below still queries the new columns:

```
var media = _mediaService.GetMediaByPath(href);
```

I'm wondering whether the best way to fix this is to move the Migration `AddPropertyTypeValidationMessageColumns` higher up in `UmbracoPlan` so it's done earlier...

## Umbraco version

I am seeing this issue on Umbraco version: 8.6

Reproduction
------------

### Bug summary

See above - note also that I've included the patches from PR #6191 to significantly speed up the migration.

### Specifics

* Umbraco 8.6.0 patched with PRs related to #6191

**Comment by nul800sebastiaan** [comment]
Hey @robertjf - I'm not sure I understand here 🙈 

As specified in the migration guide, you need to first upgrade your site to 7.14 (but preferably latest v7.15) before you even attempt to migrate the content to v8. Migrating from v6 is in no way possible before taking that step first and verifying that the site works as expected in v7.14/15 first.

**Comment by nul800sebastiaan** [comment]
And what is Umbraco 6.6.0? That version doesn't even exist? 😮😅

**Comment by robertjf** [comment]
@nul800sebastiaan I've upgraded from 4.9 via 7.7 and 7.15.3 without any problems, and then attempted to upgrade to 6.6 (started with the RC, but that was taking way too long (hours!) so patched the version 6.6 tagged in the repository with the optimisations from the mentioned Pull Request.

So although 6.6 is not yet released, for all intents and purposes I'm using that version.

**Comment by robertjf** [comment]
If it helps, the offending line of code is in
 
https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Core/Migrations/Upgrade/V_8_1_0/ConvertTinyMceAndGridMediaUrlsToLocalLink.cs#L113

**Comment by robertjf** [comment]
To get around this problem, I've moved the line

```
            To<AddPropertyTypeValidationMessageColumns>("{3D67D2C8-5E65-47D0-A9E1-DC2EE0779D6B}");
```
from line 189 of `UmbracoPlan` up to line 182 and it works, however I'm not sure this is the best way to proceed.

**Comment by robertjf** [comment]
oops, yes - just noticed the 6.6 instead of 8.6 - d'oh!

**Comment by nul800sebastiaan** [comment]
Now it makes a lot more sense, 8.6... 😅

**Comment by nul800sebastiaan** [comment]
I am unsure if this is the way to go too, but if it works for you then you should roll with it for now and keep move on to seeing if your site has got all the data you need and is working otherwise correctly. 8.6 will be out soon, after which we could look into switching those things around.

To set some expectations: it's going to take quite a while before we can get back to looking at migrations again, priorities have changed for now and we need to ask for some patience on that 🙏

**Comment by robertjf** [comment]
all good, I can get through the migration with the patched dll; although I've run into another issue once I got past that one, but that's for another report 😎

**Comment by nul800sebastiaan** [comment]
Could you just add it to the info in #6191 so it's all collected in one place and we don't have to go hunt around for issues. #6191 is an experiment just for that anyway!

**Comment by robertjf** [comment]
There's a mention there already for this issue - is that enough, or do you want a summary in a comment?

**Comment by nul800sebastiaan** [comment]
@robertjf I was more talking about the additional issue you mentioned! :-)

> I've run into another issue once I got past that one, but that's for another report 😎

**Comment by hfloyd** [comment]
I've created a pull request for this. https://github.com/umbraco/Umbraco-CMS/pull/7940

**Comment by umbrabot** [comment]
Hiya @robertjf,

Just wanted to let you know that we noticed that this issue got a bit stale and might not be relevant any more.

We will close this issue for now but we're happy to open it up again if you think it's still relevant (for example: it's a feature request that's not yet implemented, or it's a bug that's not yet been fixed). 

To open it this issue up again, you can write `@umbrabot still relevant` in  a new comment as the first line. It would be *super* helpful for us if on the next line you could let us know why you think it's still relevant.

For example:

>@umbrabot still relevant
>This bug can still be reproduced in version `x.y.z` 

This will reopen the issue in the next few hours.

Thanks, from your friendly Umbraco GitHub bot :robot: :slightly_smiling_face:
